metricflow/api/simple_api.py

```diff
@@ -0,0 +1,259 @@
+from typing import Dict, List, Optional
+
+from metricflow.configuration.config_handler import ConfigHandler
+from metricflow.configuration.constants import CONFIG_DWH_SCHEMA
+from metricflow.dataflow.sql_table import SqlTable
+from metricflow.engine.metricflow_engine import (
+    MetricFlowEngine,
+    MetricFlowExplainResult,
+    MetricFlowQueryRequest,
+    MetricFlowQueryResult,
+)
+from metricflow.engine.models import Dimension, Materialization, Metric
+from metricflow.engine.utils import build_user_configured_model_from_config, convert_to_datetime
+from metricflow.model.model_validator import ModelValidator
+from metricflow.model.objects.user_configured_model import UserConfiguredModel
+from metricflow.model.semantic_model import SemanticModel
+from metricflow.model.validations.validator_helpers import ValidationIssueType
+from metricflow.protocols.sql_client import SqlClient
+from metricflow.sql.optimizer.optimization_levels import SqlQueryOptimizationLevel
+from metricflow.sql_clients.sql_utils import make_sql_client_from_config
+
+
+class MetricFlowSimpleAPI:
+    """Simple MetricFlow Python Interface."""
+
+    def __init__(
+        self,
+        sql_client: Optional[SqlClient] = None,
+        semantic_model: Optional[SemanticModel] = None,
+        system_schema: Optional[str] = None,
+    ):
+        """If parameters not passed, build via config."""
+        handler = ConfigHandler()
+        self._sql_client = sql_client or make_sql_client_from_config(handler)
+        self._system_schema = system_schema or handler.get_value(CONFIG_DWH_SCHEMA)
+        self._semantic_model = semantic_model or SemanticModel(build_user_configured_model_from_config(handler))
+        self._mf = MetricFlowEngine(
+            semantic_model=self._semantic_model,
+            sql_client=self._sql_client,
+            system_schema=self._system_schema,
+        )
+
+    @property
+    def user_configured_model(self) -> UserConfiguredModel:  # noqa: D
+        return self._semantic_model.user_configured_model
+
+    @property
+    def mf_engine(self) -> MetricFlowEngine:  # noqa: D
+        return self._mf
+
+    @property
+    def system_schema(self) -> str:  # noqa: D
+        return self._system_schema
```

Comment:
Please add docstrings for all of these - this is going to be a public-facing developer API, so we should have full documentation of properties and what they're for.

Reply:
added